src/components/ModalWithForm.js

```diff
@@ -9,23 +9,27 @@ class ModalWithForm extends Modal {
   }
 
   close() {
-    this._modalForm.reset();
     super.close();
   }
 
+  _clearForm() {
+    this._modalForm.reset();
+  }
+
   _getInputValues() {
-    const inputList = [...this._modalForm.querySelectorAll('input')];
+    const inputList = [...this._modalForm.querySelectorAll("input")];
     const inputValues = {};
     for (const input of inputList) {
       inputValues[input.name] = input.value;
-    };
+    }
     return inputValues;
   }
 
   setEventListeners() {
     this._modalForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
+      this._clearForm();
       this.close();
```

Comment:
* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well